### PR TITLE
Add exit_hook configuration option

### PR DESCRIPTION
**Pull request by willyhun**
This is a simple addition to run a secondary script to distribute the new certificate files or restarts the configured services

### CODE REVIEW

File: pdns_api.sh [hunk 1]
```diff
@@ -295,6 +295,19 @@ soa_edit() {
   fi
 }
 
+exit_hook() {
+
+  if [[ ! -z "${PDNS_EXIT_HOOK:-}" ]]; then
+      if [[ -x "${PDNS_EXIT_HOOK}" ]]; then
+        exec ${PDNS_EXIT_HOOK}
```
Comment (silkeh) [inline pdns_api.sh]: ^-- SC2086: Double quote to prevent globbing and word splitting.

File: pdns_api.sh [hunk 2]
```diff
@@ -325,6 +338,13 @@ main() {
     exit 0
   fi
 
+  # Interface for exit_hook
+  if [[ "${hook}" = "exit_hook" ]]; then
+    shift
+    exit_hook $@
```
Comment (silkeh) [inline pdns_api.sh]: ^-- SC2068: Double quote array expansions to avoid re-splitting elements.

File: pdns_api.sh [hunk 3]
```diff
@@ -295,6 +295,19 @@ soa_edit() {
   fi
 }
 
+exit_hook() {
+
```
Comment (silkeh) [inline pdns_api.sh]: Please remove the white space at the begin/end of the function.

File: pdns_api.sh [hunk 4]
```diff
@@ -295,6 +295,19 @@ soa_edit() {
...
+  if [[ ! -z "${PDNS_EXIT_HOOK:-}" ]]; then
+      if [[ -x "${PDNS_EXIT_HOOK}" ]]; then
+        exec ${PDNS_EXIT_HOOK}
+      else
+        error "Your ${PDNS_EXIT_HOOK} not an executable"
```
Comment (silkeh) [inline pdns_api.sh]: Maybe change this to: `error "${PDNS_EXIT_HOOK} is not an executable"`

**Comment by silkeh** [comment]
Thanks! I have added some (very minor) comments to take a look at.

**Comment by willyhun** [comment]
The suggestions implemented

**Comment by silkeh** [comment]
Merged! Thanks again!

**Pull request merged**